src/main/java/com/hms_networks/americas/sc/extensions/util/RawNumberValueUtils.java

```diff
@@ -0,0 +1,313 @@
+      // Compare values
+      Short short1 = (Short) value1;
+      Short short2 = (Short) value2;
+      if (short1.shortValue() > short2.shortValue()) {
+        maxValue = short1;
+      } else {
+        maxValue = short2;
+      }
+    } else if (value1 instanceof Byte && value2 instanceof Byte) {
+      // Compare values
+      Byte byte1 = (Byte) value1;
+      Byte byte2 = (Byte) value2;
+      if (byte1.byteValue() > byte2.byteValue()) {
+        maxValue = byte1;
+      } else {
+        maxValue = byte2;
+      }
+    } else {
+      // Throw exception
+      throw new IllegalArgumentException(
+          "Cannot compare values of type "
+              + value1.getClass().getName()
+              + " and "
+              + value2.getClass().getName());
+    }
+
+    // Return max value
+    return maxValue;
+  }
+
+  /**
+   * Get and return the sum of the specified values.
+   *
+   * @param values values to sum
+   * @return sum of values
+   * @throws IllegalArgumentException if values are not of the same type
+   * @since 1.15.7
+   */
+  public static Object getValueAvg(Object[] values) {
+    // Get sum of values
+    Object sum = getValueSum(values);
+
+    // Get average of values
+    Object avg = getValueDivide(sum, values.length);
+
+    // Return average
+    return avg;
+  }
+
+  /**
+   * Get and return the sum of the specified values.
+   *
+   * @param values values to sum
+   * @return sum of values
+   * @throws IllegalArgumentException if values are not of the same type
+   * @since 1.15.7
+   */
+  public static Object getValueSum(Object[] values) {
+    // Sum values by type
+    Object sum;
+    Object firstValue = values[0];
+    if (firstValue instanceof Integer) {
+      // Sum values
+      int sumInt = 0;
+      for (int i = 0; i < values.length; i++) {
+        sumInt += ((Integer) values[i]).intValue();
+      }
+      sum = new Integer(sumInt);
+    } else if (firstValue instanceof Long) {
+      // Sum values
+      long sumLong = 0;
+      for (int i = 0; i < values.length; i++) {
+        sumLong += ((Long) values[i]).longValue();
+      }
+      sum = new Long(sumLong);
+    } else if (firstValue instanceof Float) {
+      // Sum values
+      float sumFloat = 0;
+      for (int i = 0; i < values.length; i++) {
+        sumFloat += ((Float) values[i]).floatValue();
+      }
+      sum = new Float(sumFloat);
+    } else if (firstValue instanceof Double) {
+      // Sum values
+      double sumDouble = 0;
+      for (int i = 0; i < values.length; i++) {
+        sumDouble += ((Double) values[i]).doubleValue();
+      }
+      sum = new Double(sumDouble);
+    } else if (firstValue instanceof Short) {
+      // Sum values
+      short sumShort = 0;
+      for (int i = 0; i < values.length; i++) {
+        sumShort += ((Short) values[i]).shortValue();
+      }
+      sum = new Short(sumShort);
+    } else if (firstValue instanceof Byte) {
+      // Sum values
+      byte sumByte = 0;
```

Comment:
will there be a rollover if sum of bytes>255?  Perhaps sum should always return long or double?

Reply:
Java has underflow/overflow, but it does not explicitly catch or throw an error in that event.

This implementation matches what is normally expected out of newer Java code, where the sum of two integers would return an integer. The JVM does not automatically convert types to work around overflow concerns. For example, in the following Java code, the raw type `T` would be `int` (note: Java does not actually allow raw/parameterized primitive types, this is just an example):

```
int x = 10;
int y = 10;
T sum = x + y;
```

Regardless of the values in `x` and `y`, the type for `sum`, `T` is always going to be `int`. If the integers are too large, it would be expected that they are converted *beforehand* (before being summed, or in this case, passed to the util method). This would look like 

```
int x = 10;
int y = 10;
T sum = (long) x + (long) y;
```

where `T` would in this case be the type `long`.

By changing the return type to be something other than the input type(s), the calling method is likely to run into errors as well, because it would expect the result to be of the same type passed in, and then a class cast exception would occur when trying to convert the returned `Object` to the expected type, because the type does not match.

Reply:
What do you think is a good solution? drop support for Bytes?

Reply:
As an alternative, what about creating a OverflowException?

Reply:
I don't think any changes are required. There is no reason for us to deviate from what the JVM would normally do, and this method is no different from something like
`int a = b + c;` except we are casting the types based on `instanceof` due to Java 1.4 limitations.

Regarding support for 'Byte', this likely won't receive much use but is included so that the utility class maintains support for *all* primitive types.

Reply:
> As an alternative, what about creating a OverflowException?

This could be done, but at the end of the day, dealing with overflow/underflow is out of the scope here I think.

This method is concerned with casting the types (as detected) and then summing the two number values exactly as they would normally in Java, with no extra processing or handling.

Reply:
Im not sure its fair to compare an operator to a method. Using operators you always have the ability to cast. In this method you dont provide that ability. IF you could specify the return type(/type to use for summation) i think that would be more comparable.

Reply:
The intention of this method is to act as an operator, though, with the advantage being that it hides a lot of the casting required in Java 1.4 when using primitive values in their corresponding `Object` class, i.e. `Integer`. This includes the side effects, positive or negative, from the values you put in.

I will take a further look at how the JVM handles summation with multiple values and types. If there are improvements that can be made in this method to more closely mimic the expectations/actual JVM behavior (when directly summing), I will change.

The goal here is to be able to have two or more data points from the same tag, take their values from `DataPoint#getValueObject()`, and then get the corresponding min/max/avg values without the large amounts of type detection and casting code at the connector/application level.

Reply:
A method overload was added to allow for overflow handling of the integer type. This covers byte, short, and int. If an overflow is detected and the option is true, the returned result is promoted to `Long` so it contains the true result. This is in addition to supporting the primitive operation behavior where the overflow would still return an integer result.